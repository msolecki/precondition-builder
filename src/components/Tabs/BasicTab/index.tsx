import React from 'react'
import Paper from '@material-ui/core/Paper'
import Grid from '@material-ui/core/Grid'
import Button from '@material-ui/core/Button'
import {makeStyles} from '@material-ui/core/styles'
import Activated from '../../Conditions/Activated'
import Logged from '../../Conditions/Logged'
import Location from '../../Conditions/Location'
import {ConditionInterface, NuggetConditionInterface} from '../../App/interfaces'
import PropTypes from 'prop-types'
import OperatingSystem from '../../Conditions/OperationSystem'

const useStyles = makeStyles(({spacing}) => ({
    root: {
        padding: spacing(3, 2),
    },
}))

interface BasicTabProps {
    handleAddCondition(value: ConditionInterface): void;

    handleActivated(value: boolean): void;

    activated: boolean;
    conditions: ConditionInterface[];
}

const BasicTab: React.FC<BasicTabProps> = (props: BasicTabProps): React.ReactElement => {
    const classes = useStyles()

    const [logged, setLogged] = React.useState<boolean | null>(null)
    const [system, setSystem] = React.useState<string | null>(null)
    const [radius, setRadius] = React.useState<number | null>(null)
    const [latLng, setLatLng] = React.useState<string | null>(null)
    const [nuggets, setNuggets] = React.useState<NuggetConditionInterface[] | null>(null)

    const clearState = (): void => {
        props.handleActivated(false)
        setLogged(null)
        setRadius(null)
        setLatLng(null)
        setNuggets(null)
    }

    const addCondition = (): void => {
        props.handleAddCondition({
            logged,
            system,
            location: {
                radius, latLng
            },
            nuggets,
        })
    }

    const onSave = (): void => {
        addCondition()
        clearState()
    }

    return (
        <Paper className={classes.root}>
            <form autoComplete="off">
                <Grid container spacing={3}>
                    <Grid item xs={12}>
                        <Activated onDataChange={props.handleActivated} activated={props.activated}/>
                    </Grid>
                </Grid>
                <Grid container spacing={3}>
                    <Grid item xs={6}>
                        <OperatingSystem onDataChange={setSystem} system={system}/>
                    </Grid>
                    <Grid item xs={6}>
                        <Logged onDataChange={setLogged} logged={logged}/>
                    </Grid>
                </Grid>
                <Location setRadius={setRadius} setLatLng={setLatLng} radius={radius} latLng={latLng}/>
                <Grid container spacing={3}>
                    {/*<HtmlNugget onDataChange={() => {
                        }} ids={}/>*/}
                </Grid>
                <Grid container spacing={3}>
                    <Grid item xs={4}>
                        <Button variant="contained" color="primary" fullWidth>
                            Add Condition
                        </Button>
                    </Grid>
                    <Grid item xs={4}>
                        <Button variant="contained" color="default" onClick={clearState} fullWidth>
                            Clear
                        </Button>
                    </Grid>
                    <Grid item xs={4}>
                        <Button variant="contained" color="secondary" onClick={onSave} fullWidth>
                            Save
                        </Button>
                    </Grid>
                </Grid>
            </form>
        </Paper>
    )
}

BasicTab.propTypes = {
    handleAddCondition: PropTypes.func.isRequired,
    handleActivated: PropTypes.func.isRequired,
    conditions: PropTypes.any, // TODO it should be fixed
    activated: PropTypes.bool.isRequired,
}

BasicTab.defaultProps = {
    activated: false,
    conditions: [],
}

export default BasicTab
