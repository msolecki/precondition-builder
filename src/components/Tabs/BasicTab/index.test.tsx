import * as React from 'react'
import {ReactTestRenderer} from 'react-test-renderer'
import {createShallow} from '@material-ui/core/test-utils'

import BasicTab, {BasicTabProps} from './index'

const component = (props: BasicTabProps): BasicTab => <BasicTab {...props}/>
const mockedFunction = jest.fn()

describe('<Activated />', () => {
    let shallow

    beforeAll(() => {
        shallow = createShallow()
    })

    test('empty snapshot', () => {
        const wrapper: ReactTestRenderer = shallow(component({
            handleAddCondition: mockedFunction,
            handleActivated: mockedFunction,
            htmlNuggetIds: ['1'],
            activated: true,
            conditions: [{logged: null, system: null, location: null, nuggets: null}]
        }))

        expect(wrapper).toMatchSnapshot()
    })
})
