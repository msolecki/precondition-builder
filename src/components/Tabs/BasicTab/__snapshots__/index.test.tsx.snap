// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Activated /> empty snapshot 1`] = `
<WithStyles(ForwardRef(Paper))
  className="makeStyles-root-1"
>
  <form
    autoComplete="off"
  >
    <WithStyles(ForwardRef(Grid))
      container={true}
      spacing={3}
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={12}
      >
        <Activated
          activated={true}
          onDataChange={[MockFunction]}
        />
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      container={true}
      spacing={3}
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={6}
      >
        <OperatingSystem
          onDataChange={[Function]}
          system={null}
        />
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={6}
      >
        <Logged
          logged={null}
          onDataChange={[Function]}
        />
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
    <Location
      latLng={null}
      radius={null}
      setLatLng={[Function]}
      setRadius={[Function]}
    />
    <HtmlNugget
      addNugget={[Function]}
      ids={
        Array [
          "1",
        ]
      }
    />
    <WithStyles(ForwardRef(Grid))
      container={true}
      spacing={3}
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={6}
      >
        <WithStyles(ForwardRef(Button))
          color="secondary"
          fullWidth={true}
          onClick={[Function]}
          variant="contained"
        >
          Add condition
        </WithStyles(ForwardRef(Button))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={6}
      >
        <WithStyles(ForwardRef(Button))
          color="default"
          fullWidth={true}
          onClick={[Function]}
          variant="contained"
        >
          Clear
        </WithStyles(ForwardRef(Button))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </form>
</WithStyles(ForwardRef(Paper))>
`;
